internal/guest/runtime/hcsv2/container.go

```diff
@@ -220,3 +236,19 @@ func (c *Container) GetStats(ctx context.Context) (*v1.Metrics, error) {
 func (c *Container) modifyContainerConstraints(ctx context.Context, rt guestrequest.RequestType, cc *guestresource.LCOWContainerConstraints) (err error) {
 	return c.Update(ctx, cc.Linux)
 }
+
+func (c *Container) getStatus() containerStatus {
+	val := atomic.LoadUint32((*uint32)(&c.status))
+	return containerStatus(val)
+}
+
+func (c *Container) setStatus(st containerStatus) error {
+	switch st {
```

Comment:
I don't feel like we need this switch anymore as we use a typed value for `st`. If someone really wants to they can cast something else to `containerStatus` and store it, but I don't think that's something we need to specifically guard against.